src/libgit2/status.c

```diff
@@ -355,6 +355,22 @@ int git_status_list_new(
 			goto done;
 		}
 
+		git_vector filtered_diffs = GIT_VECTOR_INIT;
+		git_vector_init(&filtered_diffs, 0, NULL);
+
+		size_t i;
+		git_diff_delta *delta;
+		// BLOCKME: If we like this solution, I'll look at filtering in place rather than creating a new vector
+		git_vector_foreach(&status->idx2wd->deltas, i, delta) {
```

Comment:
This looks right to me – when a skip-worktree index entry compared to the worktree is has a deleted status, it shouldn't be considered a delta. The fix is done here, because the implementation for idx<->workdir diff is generic and used for diffs of any iterator type.